idp/idputil/idputil.go

```diff
@@ -254,3 +269,308 @@ func CookiePathRelativeToLocation(cookiePath, location string, skipLocation bool
 	}
 	return u.Path + cookiePath
 }
+
+// JsonResponse mashals the provided to JSON and writes it
+// to the provided response writer.
+func JsonResponse(w http.ResponseWriter, data interface{}, c int) {
+	dj, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, "failed to create a json response", http.StatusInternalServerError)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(c)
+	fmt.Fprintf(w, "%s", dj)
```

Comment:
Why are you formatting a slice of bytes as a string to write to the writer? it seems like a very inefficient way to do `w.Write(dj)` note the contract for Writer requires the write to either succeed fully, or fail with a terminal error.